src/Runtime/jni/src/com/ibm/onnxmlir/OMRunner.java

```diff
@@ -41,207 +41,221 @@
+		{
+			put("b1", OMTensor.ONNX_TYPE_BOOL);
+			put("i1", OMTensor.ONNX_TYPE_INT8);
+			put("u1", OMTensor.ONNX_TYPE_UINT8);
+			put("i2", OMTensor.ONNX_TYPE_INT16);
+			put("u2", OMTensor.ONNX_TYPE_UINT16);
+			put("i4", OMTensor.ONNX_TYPE_INT32);
+			put("u4", OMTensor.ONNX_TYPE_UINT32);
+			put("i8", OMTensor.ONNX_TYPE_INT64);
+			put("u8", OMTensor.ONNX_TYPE_UINT64);
+			put("f2", OMTensor.ONNX_TYPE_FLOAT16);
+			put("f4", OMTensor.ONNX_TYPE_FLOAT);
+			put("f8", OMTensor.ONNX_TYPE_DOUBLE);
+			put("U25", OMTensor.ONNX_TYPE_STRING);
+		}
+	};
 
-    private static HashMap<String, Object> encodeTensor(OMTensor omt) throws Exception {
-	/* We need a byte[] for base64 encode but buffer.array()
-	 * does NOT work. Because the buffer is backed by JNI code
-	 * generated array, not a Java byte[] object. So base64
-	 * encode results in UnsupportedOperationException.
-	 */
-	ByteBuffer buffer = omt.getData();
-	byte[] bytes = new byte[buffer.limit()];
-	buffer.get(bytes);
-
-	String dtype = onnx2numpyType.get(omt.getDataType());
-
-	HashMap<String, Object> map = new HashMap<String, Object>();
-	map.put("buffer", Base64.getEncoder().encodeToString(bytes/*buffer.array()*/));
-	map.put("dtype", dtype);
-	map.put("shape", omt.getShape());
-	return map;
-    }
-
-    /* Model inputs are read from stdin encoded in JSON. This routine will
-     *
-     * - read JSON from stdin
-     * - decode JSON array with Jsoniter
-     * - call createTensor to create an OMTensor from each object
-     * - construct the OMTensorList to be fed into mainGraph
-     */
-    private static OMTensorList readStdin() throws Exception {
-        BufferedReader stdin =
-	    new BufferedReader(new InputStreamReader(System.in));
-	ArrayList<OMTensor> omtl = new ArrayList<OMTensor>();
-
-	JsonIterator json = JsonIterator.parse(stdin.readLine());
-	int count = 0;
-	while(json.readArray()) {
-	    Data data = json.read(Data.class);
-	    OMTensor omt = createTensor(data.buffer, data.shape, data.dtype);
-	    omtl.add(omt);
-	    count++;
+	private static final HashMap<Integer, String> onnx2numpyType = new HashMap<Integer, String>() {
+		{
+			put(OMTensor.ONNX_TYPE_BOOL, "|b1");
+			put(OMTensor.ONNX_TYPE_INT8, "|i1");
+			put(OMTensor.ONNX_TYPE_UINT8, "|u1");
+			put(OMTensor.ONNX_TYPE_INT16, numpyEndian + "i2");
+			put(OMTensor.ONNX_TYPE_UINT16, numpyEndian + "u2");
+			put(OMTensor.ONNX_TYPE_INT32, numpyEndian + "i4");
+			put(OMTensor.ONNX_TYPE_UINT32, numpyEndian + "u4");
+			put(OMTensor.ONNX_TYPE_INT64, numpyEndian + "i8");
+			put(OMTensor.ONNX_TYPE_UINT64, numpyEndian + "u8");
+			put(OMTensor.ONNX_TYPE_FLOAT16, numpyEndian + "f2");
+			put(OMTensor.ONNX_TYPE_FLOAT, numpyEndian + "f4");
+			put(OMTensor.ONNX_TYPE_DOUBLE, numpyEndian + "f8");
+			// numpy documentation: Unicode string
+			// https://numpy.org/doc/stable/reference/arrays.dtypes.html
+			put(OMTensor.ONNX_TYPE_STRING, "|U25");
```

Comment:
I also added `U25` which indicates a unicode string in numpy up to length 25. In the documentation, it says to avoid using S25 ->np.string_ because that maps to a byte string and it is zero-terminated bytes (not recommended).

Reply:
The addition is for line 101 and line 121. All of the changes in this file has to do with formatting.

Reply:
Let's focus on byte string now. You can leave an assertion for unicode string.